bool value
---
int32 errorCode
string errorDescription